### ISSUE TITLE
32bit version throws segmentation fault when running from terminal/file manager

### ISSUE THREAD

**Issue by fulalas** [celluloid-player/celluloid#660]
I'm using official 0.21 release version in Slackware current. If I just open Celluloid and then drag and drop a file to it, everything works just nice. However, if I call Celluloid from a terminal or file manager, it popus really quickly and then closes with a Segmentation fault message (in the output terminal). In x64 this issue doesn't happen.

**Comment by gnome-mpv** [comment]
I can't reproduce this with 32bit Debian unstable.

Would you be able to generate a [backtrace](https://github.com/celluloid-player/celluloid/blob/master/.github/CONTRIBUTING.md) and post it here?

**Comment by ncmprhnsbl** [comment]
using the same as OP strace: https://gist.github.com/ncmprhnsbl/d622432a58e42df70eddf12060beb0f9#file-gistfile1-txt
to clarify slightly: only when opening a file with celluloid, whether terminal or filemanager..
maybe relevant:
gtk+3-3.24.29
meson-0.58.0
ninja-1.10.2
gcc-10.3.0
mpv-0.33.0

SLKCFLAGS="-O2 -march=i586 -mtune=i686 -DNDEBUG"

**Comment by ncmprhnsbl** [comment]
using the same as OP 
backtrace: https://gist.github.com/ncmprhnsbl/96a8a7379ecd5e28efff9269eacbb7cd#file-btrace-txt
(extra term output tagged on the end)
and good measure, strace: https://gist.github.com/ncmprhnsbl/d622432a58e42df70eddf12060beb0f9#file-gistfile1-txt

**Comment by gnome-mpv** [comment]
>backtrace: https://gist.github.com/ncmprhnsbl/96a8a7379ecd5e28efff9269eacbb7cd#file-btrace-txt

I don't see any backtraces here, only regular log output. Did you run Celluloid in gdb?

From the log, it looks like something related to MPRIS. I can't see where exactly in the code that's triggering the crash without a backtrace though. Does it stop crashing if you disable MPRIS in the preferences dialog?

**Comment by ncmprhnsbl** [comment]
ah, sorry about that, might a have skimmed a little p:
backtrace (if i'm doing it right): https://gist.github.com/ncmprhnsbl/a8b4dcd242e8d6975f951d6a1d5ce7a5#file-gistfile1-txt
that's with MPRIS disabled (which also crashes)

**Comment by gnome-mpv** [comment]
Can you check if this patch fixes the problem?

```patch
From 7cc135eeb9f3e0a9e5a95f4e8b136ec34519b860 Mon Sep 17 00:00:00 2001
From: gnome-mpv <gnome-mpv@nowhere>
Date: Thu, 3 Jun 2021 16:42:06 +0700
Subject: [PATCH] view: Cast playlist-pos to gint before using it to set the
 property

celluloid_view_set_playlist_pos() accepts the position as a gint64 but
the property is an int.

Fixes #660.
---
 src/celluloid-view.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/celluloid-view.c b/src/celluloid-view.c
index 85ab788..08ad3f4 100644
--- a/src/celluloid-view.c
+++ b/src/celluloid-view.c
@@ -1789,7 +1789,7 @@ celluloid_view_update_playlist(CelluloidView *view, GPtrArray *playlist)
 void
 celluloid_view_set_playlist_pos(CelluloidView *view, gint64 pos)
 {
-	g_object_set(view, "playlist-pos", pos, NULL);
+	g_object_set(view, "playlist-pos", (gint)pos, NULL);
 }
 
 void
-- 
2.31.1
```

**Comment by fulalas** [comment]
Yes, it fixes the problem. Thank you! :)
